packages/nuxt/src/pages/module.ts

```diff
@@ -202,9 +202,17 @@ export default defineNuxtModule({
             // Avoid duplicate keys in the generated RouteNamedMap type
             const absolutePagePath = joinURL(parent.path, page.path)
 
-            // @ts-expect-error TODO: either fix types upstream or figure out another
             // way to add a route without a file, which must be possible
-            const route = addedPagePaths.has(absolutePagePath) ? parent : parent.insert(page.path, page.file)
+            const route = addedPagePaths.has(absolutePagePath)
+              ? parent
+              : /^\//.test(page.path)
+                // @ts-expect-error TODO: either fix types upstream or figure out another
+                // way to add a route without a file, which must be possible
+                ? rootPage.insert(page.path, page.file)
```

Comment:
this is the key thing - by inserting the page at the root, uvr is able to correctly map the route to the right place in the tree.

not sure if this is a bug in uvr or not, but it does resolve the issue for nuxt

Reply:
Interesting. I'll check it out locally in a bit.

Slightly related: I created this issue for the `@ts-expect-error` comments: https://github.com/posva/unplugin-vue-router/pull/607